nautobot/extras/models/change_logging.py

```diff
@@ -213,18 +213,32 @@ def get_snapshots(self):
 
         prior_change = self.get_prev_change(only=["object_data_v2", "object_data"])
 
-        if self.action != ObjectChangeActionChoices.ACTION_CREATE and prior_change is not None:
-            prechange = prior_change.object_data_v2
-            if prechange is None:
-                prechange = prior_change.object_data
+        # Populate the prechange field, create actions do not need to have a prechange field
+        if self.action != ObjectChangeActionChoices.ACTION_CREATE:
+            # Deal with the cases where we are trying to capture an object deletion/update and there is no prior change record.
+            # This can happen when the object is first created and the changelog for that object creation action is deleted.
+            if prior_change is None:
+                if self.action == ObjectChangeActionChoices.ACTION_DELETE:
+                    prechange = self.object_data_v2
+                    if prechange is None:
+                        prechange = self.object_data
+                else:
+                    prechange = pre_object_data_v2
+                    if prechange is None:
+                        prechange = pre_object_data
+            else:
+                prechange = prior_change.object_data_v2
+                if prechange is None:
+                    prechange = prior_change.object_data
```

Comment:
Would it be simpler (and maybe more readable) to do something like:

```suggestion
            if prior_change is not None:
                prechange = prior_change.object_data_v2 or prior_change.object_data
            elif self.action == ObjectChangeActionChoices.ACTION_DELETE:
                prechange = self.object_data_v2 or self.object_data
            else:
                prechange = pre_object_data_v2 or pre_object_data
```